### PR TITLE
Suspension

### PR DESCRIPTION
**Pull request by muukii**
suspend current subscription
resume it by emitting the latest value immediately

### CODE REVIEW

File: Sources/Verge/Library/StoreSubscription.swift [hunk 1]
```diff
@@ -84,4 +86,162 @@ public final class StoreSubscription: Hashable, Cancellable {
   }
 }
 
+public final class StoreStateSubscription: Hashable, Cancellable, @unchecked Sendable {
```
Comment (muukii) [inline Sources/Verge/Library/StoreSubscription.swift:89]: For now it's copied

File: Sources/Verge/Store/Store.swift [hunk 2]
```diff
@@ -552,38 +552,74 @@ Mutation: (%@)
...
+    if keepsAliveForSubscribers {
+      return .init(cancellable, storeCancellable: storeLifeCycleCancellable, onAction: onAction)
+        .associate(store: self) // while subscribing its Store will be alive
+    } else {
+      return .init(cancellable, storeCancellable: storeLifeCycleCancellable, onAction: onAction)
+    }
     
+  }
+
+  private func _base_primitive_sinkState(
```
Comment (muukii) [inline Sources/Verge/Store/Store.swift:595]: separated to return source cancellable

File: Sources/Verge/Store/Store.swift [hunk 3]
```diff
@@ -552,38 +552,74 @@ Mutation: (%@)
...
+
+    let onAction: (StoreStateSubscription, StoreStateSubscription.Action) -> Void = { [weak self] object, action in
+      
+      guard let self else {
+        return
+      }
+      
+      switch action {
+      case .suspend:
+        object.cancelSubscription()
```
Comment (muukii) [inline Sources/Verge/Store/Store.swift:575]: cancel while suspending

File: Sources/Verge/Store/Store.swift [hunk 4]
```diff
@@ -552,38 +552,74 @@ Mutation: (%@)
...
+      case .resume:
+        let newCancellable = _base_primitive_sinkState(
+          dropsFirst: false, // emits current value from beginning.
+          queue: queue,
+          receive: receive
+        )
+        object.replace(cancellable: consume newCancellable)
```
Comment (muukii) [inline Sources/Verge/Store/Store.swift:582]: on resume, recreate subscription

**Review by JohnEstropia — APPROVED** [review]
The approach looks good, but I think we now have to be careful that we don't miss events while the sink is suspended. Example:
- Action logs from screens presented from the suspended cell
- Prefetch requests


There are also the cases where events are fired during resume that may change some expected behavior:
- Action logs being sent when a cell reappears
- Prefetch requests being fired later than expected

**Comment by muukii** [comment]
one test failed. I'll look into it.

**Pull request merged**